Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp

```diff
@@ -126,7 +126,7 @@ bool getVideoSizeAndFormatFromCaps(const GstCaps* caps, WebCore::IntSize& size,
 
         size.setWidth(width);
         size.setHeight(height);
-    } else {
+    } else if (!gst_structure_has_name(structure, "video/x-raw") || gst_structure_has_name(structure, "format")) {
```

Comment:
In which case did you see a structure named `format`?

Reply:
The failure comes from [format being missing from here](https://github.com/GStreamer/gstreamer/blob/1.20.2/subprojects/gst-plugins-base/gst-libs/gst/video/video-info.c#L410) so checking `gst_structure_has_name(structure, "format")` guards against `gst_video_info_from_caps` being called in cases where `gst_video_info_from_caps` will fail due to not having `format`.

Reply:
`gst_structure_get_string` fetches a field of the struct, not its name.

Reply:
Yeah, just noticed that, changed to `gst_structure_has_field`.